internal/security/secretstore/tokenprovider_test.go

```diff
@@ -18,78 +18,123 @@ package secretstore
 
 import (
 	"context"
+	"errors"
+	"os"
+	"os/exec"
 	"testing"
 
 	"github.com/edgexfoundry/edgex-go/internal/security/secretstoreclient"
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
 )
 
 func TestInvalidProvider(t *testing.T) {
 	config := secretstoreclient.SecretServiceInfo{
 		TokenProvider:     "does-not-exist",
 		TokenProviderType: OneShotProvider,
 	}
-	cancel, err := testCommon(config)
+	mockExecRunner := mockExecRunner{}
+	mockExecRunner.On("LookPath", "does-not-exist").Return("", errors.New("fake file does not exist"))
+	cancel, err := testCommon(config, &mockExecRunner)
 	defer cancel()
 	assert.NotNil(t, err)
+	mockExecRunner.AssertExpectations(t)
 }
 
 func TestInvalidProviderType(t *testing.T) {
 	config := secretstoreclient.SecretServiceInfo{
-		TokenProvider:     "/bin/true",
+		TokenProvider:     "success-executable",
 		TokenProviderType: "simple",
 	}
-	cancel, err := testCommon(config)
+	mockExecRunner := mockExecRunner{}
+	cancel, err := testCommon(config, &mockExecRunner)
 	defer cancel()
 	assert.Error(t, err)
+	mockExecRunner.AssertExpectations(t)
 }
 
 func TestNoConfig(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
-	p := NewTokenProvider(ctx, logger.MockLogger{})
+	p := NewTokenProvider(ctx, logger.MockLogger{}, ExecWrapper{})
+	// don't call SetConfiguration()
 	err := p.Launch()
 	defer cancel()
 	assert.Error(t, err)
 }
 
-func TestTrue(t *testing.T) {
+func TestSuccess(t *testing.T) {
 	config := secretstoreclient.SecretServiceInfo{
-		TokenProvider:     "/bin/true",
+		TokenProvider:     "success-executable",
 		TokenProviderType: OneShotProvider,
+		TokenProviderArgs: []string{"arg1", "arg2"},
 	}
-	cancel, err := testCommon(config)
+	mockExecRunner := mockExecRunner{}
+	mockCmd := mockCmd{}
+	mockExecRunner.On("LookPath", "success-executable").Return("success-executable", nil)
+	mockExecRunner.On("CommandContext", mock.Anything, "success-executable", []string{"arg1", "arg2"}).Return(&mockCmd)
```

Comment:
Can you fold this line out so it isn't so long?

Reply:
foled